src/views/checkups/self-validation/details/components/DownloadResultsButton.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+
+import {
+  IoK8sApiBatchV1Job,
+  IoK8sApiCoreV1ConfigMap,
+  IoK8sApiCoreV1PersistentVolumeClaim,
+} from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { kubevirtConsole } from '@kubevirt-utils/utils/utils';
+import { Button, ButtonVariant, FlexItem } from '@patternfly/react-core';
+import { DownloadIcon } from '@patternfly/react-icons';
+
+import { CheckupsStatus, getJobStatus } from '../../../utils/utils';
+import {
+  createDetailedResultsViewer,
+  getConfigMapStatus,
+  getTimestampFromJob,
+  waitForNginxServer,
+} from '../../utils';
+
+type DownloadResultsButtonProps = {
+  configMap: IoK8sApiCoreV1ConfigMap;
+  jobs: IoK8sApiBatchV1Job[];
+  onError?: (error: null | string, url?: null | string) => void;
+  pvcs: IoK8sApiCoreV1PersistentVolumeClaim[];
+};
+
+const DownloadResultsButton: React.FC<DownloadResultsButtonProps> = ({
+  configMap,
+  jobs,
+  onError,
+  pvcs,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isCreatingViewer, setIsCreatingViewer] = useState(false);
+  const [isWaitingForServer, setIsWaitingForServer] = useState(false);
+
+  const job = jobs?.[0];
```

Comment:
Why do we need to pass the array of jobs if we only use the first one? Can we pass only the first one?